bin/inference/pycbc_inference_extract_samples

```diff
@@ -101,14 +116,49 @@ if len(opts.input_file) == 1:
 # convert samples to a dict in which the keys are the labels
 # also stack results if multiple files were provided
 if len(opts.input_file) > 1:
-    samples = {labels[p]: numpy.concatenate([s[p] for s in samples])
-               for p in params}
+    if opts.combine_via_sampling:
+        raw_samples = {labels[p]: numpy.concatenate([s[p] for s in samples])
+                   for p in params}
+        logz_list = []
+        dlogz_list = []
+        len_list = []
+        raw_samps_list = []
+        weights_list = []
+        # get samples and parameters from each file
+        for file in opts.input_file:
+            fp = loadfile(file, 'r')
+            logz, dlogz = fp.log_evidence
```

Comment:
Not all samplers write out evidence (`emcee`, for example). If not, trying to get the evidence will raise a KeyError (see [here](https://github.com/gwastro/pycbc/blob/e88c2725d0bc87b9ae977e29f3069c4abf310032/pycbc/inference/io/base_hdf.py#L437-L442) ). Put a try/except around this. In the case that there's no log evidence you'll need to have if statements below whenever you deal with evidence.